core/node/consensus/src/storage/vm_reader.rs

```diff
@@ -0,0 +1,341 @@
+use std::time::Duration;
+
+use zksync_concurrency::ctx::Ctx;
+use zksync_contracts::load_contract;
+use zksync_node_api_server::{execution_sandbox::BlockStartInfo, tx_sender::TxSender};
+use zksync_system_constants::DEFAULT_L2_TX_GAS_PER_PUBDATA_BYTE;
+use zksync_types::{
+    api::BlockId,
+    ethabi::{Address, Contract, Function, Token},
+    fee::Fee,
+    l2::L2Tx,
+    transaction_request::CallOverrides,
+    Nonce, U256,
+};
+
+use crate::storage::ConnectionPool;
+
+pub struct VMReader {
+    pool: ConnectionPool,
+    tx_sender: TxSender,
+    consensus_authority_contract: Contract,
+    validator_registry_address: Option<Address>,
+    validator_registry_contract: Contract,
+    attester_registry_address: Option<Address>,
+    attester_registry_contract: Contract,
```

Comment:
Do we need 3 dedicated contracts for consensus? Our contracts are already very sophisticated, so my fear is that it would complicate them even more.
I do not have concrete proposals, but I strongly suggest discussing the contracts design with @vladbochok and @StanislavBreadless 

Not actionable in this PR I guess.

Reply:
That's my fault. I did the design for the contracts and thought that smaller contracts would be preferred. They can be merged into a single larger contract though.